src/debuggability.js

```diff
@@ -156,7 +156,9 @@ var fireDomEvent = (function() {
             return function(name, event) {
                 var domEvent = new CustomEvent(name.toLowerCase(), {
                     detail: event,
-                    cancelable: true
+                    cancelable: true,
+                    reason: event.reason,
+                    promise: event.promise
```

Comment:
Would it make sense to write this as `get promise() { return event.promise; }`? I'm unfamiliar with the compilation pipeline for bluebird, and whether that syntax is supported on all the browsers Bluebird targets.

If we wanted to be even more protective, we could do something like:
```js
var reason = event.reason;
var promise = event.promise;
var domEvent = new CustomEvent(name.toLowerCase(), {
  detail: event,
  cancelable: true,
  get reason() { return reason; },
  get promise() { return promise; }
})
```

or the equivalent with `Object.defineProperty`

Let me know if that's a desirable change & I can update the PR

Reply:
Bluebird actually targets ES3 browsers too (supports IE7 for example) - on the other hand this branch of code isn't reached on these browsers anyway.